src/server/server_family.cc

```diff
@@ -76,6 +93,55 @@ ABSL_DECLARE_FLAG(bool, tls);
 ABSL_DECLARE_FLAG(string, tls_ca_cert_file);
 ABSL_DECLARE_FLAG(string, tls_ca_cert_dir);
 
+bool AbslParseFlag(std::string_view in, ReplicaOfFlag* flag, std::string* err) {
+#define RETURN_ON_ERROR(m)                                               \
+  do {                                                                   \
+    *err = m;                                                            \
+    LOG(WARNING) << "Error in parsing arguments for --replicaof: " << m; \
+    return false;                                                        \
+  } while (0)
+
+  if (in.empty()) {  // on empty flag "parse" nothing. If we return false then DF exists.
+    *flag = ReplicaOfFlag{};
+    return true;
+  }
+
+  auto pos = in.find_last_of(':');
+  if (pos == string::npos)
```

Comment:
I could swear that I wrote this somewhere, but now I can't find it :|
Anyway, I'd change the `RETURN_ON_ERROR` macro to accept a condition (i.e. 2 input "variables") and call it, and have the `if` inside.
For example `RETURN_ON_ERROR(pos == string::npos, "missing ':'.);`
This is also better with your naming, `_ON_ERROR` reads to me that it only returns upon encountering an error :)

Reply:
okay sure! I probably missed that